### ISSUE TITLE
Recent composer changes break dependency detection.

### ISSUE THREAD

**Issue by jmalloc** [eloquent/composer-npm-bridge#8]
**Symptom**

`composer install` does not attempt install NPM modules at all.

**Cause**

[`NpmBridge::isDependantPackage()`](https://github.com/eloquent/composer-npm-bridge/blob/develop/src/NpmBridge.php#L147) is unable to detect `eloquent/composer-npm-bridge` as a dependency when composer is installing dependencies based on a lock file.

Recent changes in composer/composer@c9cff6387a874e15e191119349a4f33c488b679d means that all  dependencies are cleared from the `RootPackage` object when installing from a lock file. Unfortunately this is the same package object made available to the plugin API.

It looks like the change was made in order to prevent an issue where dependencies specified in the composer.json file take precedence over those in the lock file.

**Solution**

We'll probably need to talk to the Composer guys about whether this change was a quick-fix or whether it's likely to stay this way.  If it's likely to stay I guess we'll need to parse the composer.json file ourselves (yuck). As for "fixing" this in Composer - I don't know what the best approach would be to keeping these required packages available to the Composer API only without causing a regression of the issue original issue.

**Workaround**

We'll use composer install/update scripts as a stop gap.

**Comment by jmalloc** [comment]
FYI @koden-km

**Comment by RafaelKa** [comment]
Detect and read packages composer.json file and get required packages from this file.

**Comment by ezzatron** [comment]
@RafaelKa In order for this plugin to work it needs to know which composer packages depend on `eloquent/composer-npm-bridge`, and that information is not in `composer.json`. It's possible that the lock file could be used I suppose, but I don't know if it gets written in time for a plugin to use it.

**Comment by RafaelKa** [comment]
@ezzatron i mean not the "root" composer.json file, i mean composer.json from each installed/updated package which should be checked if eloquent/composer-npm-bridge is required!

**Comment by cs278** [comment]
Unless I am missing something this plugin works perfectly fine, at least it does what I expect it to. Looks like the problem with Composer has been fixed ages ago, I suspect [this commit](https://github.com/composer/composer/commit/b4698568d2575a3f3d0851a38d27cf8a5cccb706) fixed it.

Installing from lock file:

```
Installing NPM dependencies for Composer dependencies
Installing NPM dependencies for xxx/yyyy
clean-css@3.1.9 node_modules/clean-css
├── commander@2.6.0
└── source-map@0.1.43 (amdefine@1.0.0)

hogan.js@3.0.2 node_modules/hogan.js
├── mkdirp@0.3.0
└── nopt@1.0.10 (abbrev@1.0.7)
```

**Comment by ezzatron** [comment]
Looks like you could be right! I'll take a look as soon as I'm able, and if everything looks good, I'll re-open #10 and sort it out too.

**Comment by ezzatron** [comment]
Fixed in [3.0.0](https://github.com/eloquent/composer-npm-bridge/releases/tag/3.0.0).
